pagopa/io-app
# chore: [IOBP-236] Banner pay with CIE into IDPay discount initiative

## ⚠️ This PR depends on #4983 and #4995

## Short description
This PR adds a `Banner` inside a discount initiative details if the user has already a CIE onboarded to his account. The banner is shown from the first time the user lands in the bonus detail, and remains visible until:
1. The user closes it via "X" close button
2. The user enables the initiative (either via banner or via toggle within the relevant section)

## List of changes proposed in this pull request
- Added a new action preferences to dismiss the banner, the state is persisted on the device based on the initiativeId;
- Added a new component `IdPayCodeCieBanner` where is handled all the business logic to show or not the banner

## How to test
Open a discount initiative, the first time you open the detail you should see the banner. If you dismiss it, even restarting the app you should not see it anymore.

## Preview
https://github.com/pagopa/io-app/assets/34343582/01245e66-b80f-4f72-94f9-5a7b87b37c1c
